03_cross_cancer_classification/plot_flip_labels_results.ipynb

```diff
@@ -0,0 +1,546 @@
+      "text/plain": [
+       "<Figure size 1080x576 with 6 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "plot_genes = flip_norm_df.gene.unique()\n",
+    "sns.set({'figure.figsize': (15, 8)})\n",
+    "fig, axarr = plt.subplots(2, 3)\n",
+    "for ix, plot_gene in enumerate(plot_genes):\n",
+    "    ax = axarr[ix // 3, ix % 3] \n",
+    "    plot_df = flip_norm_df[flip_norm_df.gene == plot_gene]\n",
+    "    sns.pointplot(data=plot_df, x='percent_flip', y='aupr', hue='identifier', ax=ax)\n",
+    "    ax.set_xlabel('Percent of 1 labels flipped to 0')\n",
+    "    ax.set_ylabel('Change in AUPR vs. negative control')\n",
+    "    ax.set_title(plot_gene)\n",
+    "    ax.get_legend().remove()\n",
+    "    ax.axhline(y=0, color='black', linestyle='--')\n",
+    "plt.tight_layout()\n",
```

Comment:
Also the results seem to be bimodal for a couple genes (PTEN and KRAS). Is there a reason for different cancer types to be more or less able to differentiate between mutated and not? Are PTEN and KRAS passenger mutations instead of drivers in the cancer types that perform less well?

Reply:
> Is there supposed to be a plot in the bottom right subplot?

Nope, there's just an odd number of subplots so I have an empty grid section. I'll get rid of the unused axes so this is clearer.

> Also the results seem to be bimodal for a couple genes (PTEN and KRAS). Is there a reason for different cancer types to be more or less able to differentiate between mutated and not? Are PTEN and KRAS passenger mutations instead of drivers in the cancer types that perform less well?

This is a good question that I've been thinking about myself. My guess isn't necessarily that they're not drivers in these cancer types, but rather that they're very rarely mutated in the dataset that I'm using (very imbalanced labels) in these cancer types, so classification doesn't work well. I've been thinking about doing an analysis that looks at label imbalance vs. performance in more detail, so stay tuned on that.

Your thought (they may just not be drivers in these cancer types) is definitely a possibility as well; some drivers are definitely thought to be cancer type-specific. Both PTEN and KRAS are thought to have fairly ubiquitous driver effects, though - looking at Figure 2 from [this paper from TCGA](https://doi.org/10.1016/j.cell.2018.02.060), it looks like they're both predicted to be drivers in multiple cancer types, including some of the cancers that perform poorly in my analyses. I'm pretty sure their driver prediction analysis primarily just uses mutation frequency in DNA sequencing data, so they may be better powered to detect driver effects in these cancer types than my analysis which requires paired DNA/RNA (although I'm not super familiar with their methods so I'm guessing a bit at this point). Definitely interesting stuff to think about!